test/smoketest/requirements.in

```diff
@@ -0,0 +1,8 @@
+# "Abstract" dependencies for the project
+#
+# The sibling requirements.txt file contains the corresponding concrete /
+# pinned dependencies.  You can use pip-tools to automatically update
+# requirements.txt from this file:
+# https://github.com/jazzband/pip-tools
```

Comment:
Interesting.  But I'm normally reluctant to introduce new requirements. Can you contrast this with other options for maintaining requirements.txt?